misp:Feature - v3.0.0 Improve the "add attribute" functionality

bliew-splunk opened the pull request:
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

## Pull Request Checklist

#### Please check if your PR fulfills the following requirements:
- [X] Testing of all the changes has been performed (for bug fixes / features)
- [X] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [X] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [X] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [X] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [X] Do NOT submit updates to dependencies unless it fixes an issue.
## Pull Request Type

#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [X] If this is a new connector or you are adding new actions
    - [X] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.



## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- Please see `release_notes/3.0.0.md`

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.
- Please see `release_notes/3.0.0.md`. I have added 2 new attribute related actions. The existing functionality of create_event action to add attributes was not working.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

hardik-crest commented:
Hello @bliew-splunk ,
I have started working on the PR u have raised for MISP : https://github.com/splunk-soar-connectors/misp/pull/11.

**We noticed some changes in the latest update and have a couple of questions for clarification:**

- Previously, the create_event action included an `add_attribute` functionality. This functionality seems to be missing now.

- The `add_attribute` function itself was previously called `update_event`. It appears you've renamed it.

Given that adding an attribute essentially updates the event, the original `update_event` action name seems appropriate. To avoid backward compatibility issues, we suggest keeping `update_event` unless there's a strong reason for the change to `add_attribute`.

bliew-splunk commented:
> Hello @bliew-splunk , I have started working on the PR u have raised for MISP : #11.
> 
> **We noticed some changes in the latest update and have a couple of questions for clarification:**
> 
> * Previously, the create_event action included an `add_attribute` functionality. This functionality seems to be missing now.
> * The `add_attribute` function itself was previously called `update_event`. It appears you've renamed it.
> 
> Given that adding an attribute essentially updates the event, the original `update_event` action name seems appropriate. To avoid backward compatibility issues, we suggest keeping `update_event` unless there's a strong reason for the change to `add_attribute`.

@hardik-crest The MISP API separates the functionality of updating an event and adding attributes:
- https://www.misp-project.org/openapi/#tag/Events/operation/editEvent
- https://www.misp-project.org/openapi/#tag/Attributes/operation/addAttribute

I have indeed separated out the attribute functionality from `create_event` into two new actions `add attribute` and `bulk add attributes`.
* Added new action `add attribute` to add a single attribute to an existing event.
* Added new action `bulk add attributes` to add multiple attributes to an existing event.

I was engaged by a sales engineer to update this app, because the attribute functionality was not working for a customer.
I believe that separating this out makes it easier for the user to understand.

I understand that this would be a breaking change for the users, but I don't have the time to work further on this.

tapishj-splunk commented on misp_connector.py:
```diff
@@ -225,9 +227,10 @@ def initialize(self):
         config = self.get_config()
         auth = {"Authorization": config.get("api_key")}
+
+        self.debug_print("making rest call")
```
🪓

bliew-splunk commented:
Hi @tapishj-splunk, I understand you're now assigned to https://splunk.atlassian.net/browse/PAPP-33460.
Would it be better to follow the approach @hardik-crest has described here: https://github.com/splunk-soar-connectors/misp/pull/11#issuecomment-2057478533, to preserve backwards compatibility?
The current code changes are a breaking change for existing users.

tapishj-splunk commented:
> Hi @tapishj-splunk, I understand you're now assigned to https://splunk.atlassian.net/browse/PAPP-33460. Would it be better to follow the approach @hardik-crest has described here: [#11 (comment)](https://github.com/splunk-soar-connectors/misp/pull/11#issuecomment-2057478533), to preserve backwards compatibility? The current code changes are a breaking change for existing users.

Hi @bliew-splunk sorry for the late reply. Ideally we shouldn't be making changes that break customer playbooks. Even with Hardik's suggestions playbooks would be broken because the input parameters would be broken. I think with the create_event action it makes sense to keep the parameters you removed and add the bulk_add_attributes. Not sure if we want to make those changes to update_event, unless this is an ask by several customers or if it's a major problem. What are your thoughts?

bliew-splunk commented:
> > Hi @tapishj-splunk, I understand you're now assigned to https://splunk.atlassian.net/browse/PAPP-33460. Would it be better to follow the approach @hardik-crest has described here: [#11 (comment)](https://github.com/splunk-soar-connectors/misp/pull/11#issuecomment-2057478533), to preserve backwards compatibility? The current code changes are a breaking change for existing users.
> 
> Hi @bliew-splunk sorry for the late reply. Ideally we shouldn't be making changes that break customer playbooks. Even with Hardik's suggestions playbooks would be broken because the input parameters would be broken. I think with the create_event action it makes sense to keep the parameters you removed and add the bulk_add_attributes. Not sure if we want to make those changes to update_event, unless this is an ask by several customers or if it's a major problem. What are your thoughts?

Thanks Tapish, I've messaged you on Slack
